### PR TITLE
[c2cpg] Defer method decl handling

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
For method de-duplication (CDT always copy methods from header files into the implementation files so we would end up generating CPG METHODs for both with the same fullName), we now skip over all method decls but remember their properties.
Afterwards, a new pass stubs them iff no actual method definition exists.

### CODE REVIEW

**Comment by max-leuthaeuser** [comment]
FYI @ml86 You may want to have a look already as it is working fine.

File: joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/CGlobal.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    lineNumber: Option[Int],
+    columnNumber: Option[Int],
+    lineNumberEnd: Option[Int],
+    columnNumberEnd: Option[Int],
+    signature: String,
+    offset: Option[(Int, Int)],
+    parameter: Seq[ParameterInfo],
+    modifier: Seq[String]
+  )
+  final class ParameterInfo(
```
Comment (ml86) [inline joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/CGlobal.scala:24]: Is there a reason why MethodInfo is a case class and ParameterInfo is not?

**Review by ml86 — APPROVED** [review]

File: joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/CGlobal.scala (hunk 1 above)
Comment (max-leuthaeuser) [reply joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/CGlobal.scala:24]: ParameterInfo has two mutable fields (to set var arg info). Scala case classes should never be mutable as it breaks some contracts when accessing them in a parallel environment (says the Scala doc).

Comment (ml86) [reply joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/CGlobal.scala:24]: Missed the mutability part. Makes sense.

**Pull request merged**